src/meta/meta_ckpt.c

```diff
@@ -518,6 +518,166 @@ __wt_meta_ckptlist_get(
     return (ret);
 }
 
+#ifdef HAVE_DIAGNOSTIC
+/*
+ * __assert_ckpt_matches --
+ *     Assert that given two checkpoints match.
+ */
+static void
+__assert_ckpt_matches(WT_SESSION_IMPL *session, WT_CKPT *ckpt_a, WT_CKPT *ckpt_b)
+{
+    if (ckpt_a->order != ckpt_b->order)
+        WT_ASSERT(session, false);
```

Comment:
Yes, it is unusual, because I was returning a boolean from this function before, "is there a match .. true/false". But then while debugging, I wanted to make it easy to catch in gdb and restructured it. This will obviously change in the final version of the change. Similarly with refactoring, eventually I will have less code duplication.

For now, I raised this PR to share what I am up to and get buy-in with the approach.